### PR TITLE
Add vrrp Python plugin

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

File: collectd/files/plugin/collectd_vrrp.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collectd
+
+import collectd_base as base
+
+from pyroute2 import IPRoute
```
Comment (elemoine) [inline collectd/files/plugin/collectd_vrrp.py:20]: Does this require adding `python-pyroute2` to the list of packages?

File: collectd/files/plugin/collectd_vrrp.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        super(VrrpPlugin, self).config_callback(conf)
+
+        for node in conf.children:
+            if node.key == 'IPAddress':
+                item = {}
+                for child_node in node.children:
+                    if child_node.key not in ('address', 'label'):
+                        continue
+                    item[child_node.key] = child_node.values[0]
+                if not item.get('address'):
```
Comment (elemoine) [inline collectd/files/plugin/collectd_vrrp.py]: Super minor, and quite personal, but I'd find this more elegant:

``` py
    if 'address' not in item:
```

File: collectd/files/plugin/collectd_vrrp.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                self.ip_addresses.append(item)
+
+        if len(self.ip_addresses) == 0:
+            self.logger.error("vrrp: Missing 'IPAddress' parameter")
+
+    def itermetrics(self):
+        for ip_address in self.ip_addresses:
+            v = 1 if self.ipr.get_addr(address=ip_address['address']) else 0
+            data = {'values': v, 'meta': {'ip_address': ip_address['address']}}
+            if ip_address.get('label'):
```
Comment (elemoine) [inline collectd/files/plugin/collectd_vrrp.py]: Ditto.

``` py
    if 'label' in ip_address:
```

**Review by elemoine — CHANGES_REQUESTED** [review]
A few comments, otherwise looks good to me.

File: collectd/files/plugin/collectd_vrrp.py (hunk 1 above)
Comment (simonpasquier) [reply collectd/files/plugin/collectd_vrrp.py:20]: this is done in the keepalived formula.

File: collectd/files/plugin/collectd_vrrp.py (hunk 3 above)
Comment (simonpasquier) [reply collectd/files/plugin/collectd_vrrp.py]: done

File: collectd/files/plugin/collectd_vrrp.py (hunk 2 above)
Comment (simonpasquier) [reply collectd/files/plugin/collectd_vrrp.py]: done

**Review by elemoine — APPROVED** [review]

**Pull request merged**